woocommerce/woocommerce-gutenberg-products-block
# Update the sidebar notice we show for incompatible extensions

Fixes #10855

## What

This PR introduces the following two changes:

- Change the wording of the incompatibility sidebar notice from
_"The following extension **is incompatible** with the block-based checkout."_ to 
_"The following extensions **may be incompatible** with the block-based checkout."_.
- Display all extensions that explicitly declared incompatibility with the Cart and Checkout blocks.

## Why

### Enhance clarity and user perception:

The adjustment in the wording from “is incompatible” to “might be incompatible” aims to clarify potential impacts and reduce unnecessary user concern over extension incompatibilities, by indicating a possibility rather than a certainty of conflict.

### Promote informed decision-making:

By displaying all extensions explicitly declaring incompatibility, we enable users to make informed decisions on extension use, reducing unforeseen disruptions and enhancing user trust and experience.

### Encourage proactive conflict resolution:

Clear, comprehensive incompatibility information empowers users and extension developers to proactively address conflicts, likely reducing support requests and fostering a harmonious ecosystem.

### Optimize user experience:

These changes are part of our ongoing efforts to minimize friction and promote understanding, enabling users to navigate the system with greater confidence and ease.

> **Note**
> The "Why" section had been created using GPT-4.

## Testing Instructions

### Test updated text

1. Install and activate the [Coinbase Commerce extension](https://wordpress.org/plugins/coinbase-commerce/).
2. Install and activate the [Paystack WooCommerce Payment Gateway extension](https://wordpress.org/plugins/woo-paystack/).
3. Create a test page and add the Checkout block to it.
4. Open the settings sidebar, if not already open.
5. Verify that the incompatibility notice lists the `Coinbase` and `Paystack` extensions:
<img width="280" alt="Screenshot 2023-09-22 at 19 57 49" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/f2779b98-0283-48fb-a1e5-fd43aa82bb46">

> **Note**
> If the compatibility notice does not appear, then open the inspector, go to `Application » Local Storage`, select your local site, remove the entry `wc-wc-blocks_dismissed_incompatible_payment_gateways_notices`, and refresh the page:
<img width="1279" alt="Screenshot 2023-09-22 at 19 58 40" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/0d34d3c9-9330-42cd-8242-1bb95a08ba1e">

### Test display of incompatible extensions

1. Install and activate the following three helper plugins:
    - [helper-plugin-1.zip](https://github.com/woocommerce/woocommerce-blocks/files/12701036/helper-plugin-1.zip)
    - [helper-plugin-2.zip](https://github.com/woocommerce/woocommerce-blocks/files/12701039/helper-plugin-2.zip)
    - [helper-plugin-3.zip](https://github.com/woocommerce/woocommerce-blocks/files/12701041/helper-plugin-3.zip)
2. Create a test page and add the Checkout block to it.
4. Open the settings sidebar, if not already open.
5. Verify that the incompatibility notice lists the `A → Incompatible Extension`, `Coinbase`, `N → Incompatible Extension
`, `Paystack` and `Z → Incompatible Extension` extensions:
<img width="277" alt="Screenshot 2023-09-22 at 20 12 20" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/f86e5ce1-6a1b-4ffa-8c24-a8b88fabfe7a">

### Test closing of sidebar notice

1. Ensure that you executed the previous test case.
2. Dismiss the incompatibility sidebar notice by clicking on the X in the upper-right corner.
3. Refresh the page and verify that the incompatibility sidebar notice remains hidden.


## Screenshots or screencast

<table>
<tr>
<td valign="top">Before:
<br><br>
<img width="279" alt="Screenshot 2023-09-22 at 20 23 22" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/c8be0e82-fddc-46f2-a8f7-eabda60e3d31">
</td>
<td valign="top">After:
<br><br>
<img width="277" alt="Screenshot 2023-09-22 at 20 12 20" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/f86e5ce1-6a1b-4ffa-8c24-a8b88fabfe7a">
</td>
</tr>
</table>


## WooCommerce Visibility


## Checklist

Required:

Conditional:

## Changelog

> Update: Adjust text of incompatibility sidebar notice and show extensions, that explicitly declared incompatibility with the Cart and Checkout blocks.
